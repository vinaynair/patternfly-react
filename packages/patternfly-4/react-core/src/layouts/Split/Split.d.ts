import { SFC, HTMLProps, ReactType } from 'react';
import { OneOf } from '../../typeUtils';
import { GutterSize, getGutterModifier } from '../../styles/gutters';

export interface SplitProps extends HTMLProps<HTMLDivElement> {
  component?: ReactType<SplitProps>;
  gutter?: OneOf<typeof GutterSize, keyof typeof GutterSize>;
}

declare const Split: SFC<SplitProps>;

export default Split;
