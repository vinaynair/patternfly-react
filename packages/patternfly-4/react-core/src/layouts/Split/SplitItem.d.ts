import { SFC, HTMLProps } from 'react';

export interface SplitItemProps extends HTMLProps<HTMLDivElement> {
  isMain: boolean;
}

declare const SplitItem: SFC<SplitItemProps>;

export default SplitItem;
