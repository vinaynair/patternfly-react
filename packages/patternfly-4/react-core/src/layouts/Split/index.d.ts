export { default as Split, SplitProps } from './Split';
export { default as SplitItem, SplitItemProps } from './SplitItem';
