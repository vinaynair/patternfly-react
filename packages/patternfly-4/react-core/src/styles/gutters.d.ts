import { StyleDeclarationStatic, getModifier } from '@patternfly/react-styles';
import { OneOf } from '../typeUtils';

export const GutterSize: {
  sm: 'sm';
  md: 'md';
  lg: 'lg';
};

export function getGutterModifier(
  styleObj: any,
  gutter: OneOf<typeof GutterSize, keyof typeof GutterSize>,
  defaultValue: StyleDeclarationStatic | string
): string | StyleDeclarationStatic;
