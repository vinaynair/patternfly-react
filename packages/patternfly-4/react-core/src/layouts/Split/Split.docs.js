import { Split, SplitItem, GutterSize } from '@patternfly/react-core';
import Simple from './examples/SimpleSplit';
import WithGutter from './examples/SplitWithGutter';

export default {
  title: 'Split',
  components: {
    Split,
    SplitItem
  },
  enumValues: {
    'Object.keys(GutterSize)': Object.keys(GutterSize)
  },
  examples: [Simple, WithGutter]
};
