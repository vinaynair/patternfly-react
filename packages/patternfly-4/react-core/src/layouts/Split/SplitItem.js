import React from 'react';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/layouts/Split/styles.css';
import { css } from '@patternfly/react-styles';

const propTypes = {
  isMain: PropTypes.bool,
  children: PropTypes.node,
  className: PropTypes.string
};

const defaultProps = {
  isMain: false,
  className: '',
  children: null
};

const SplitItem = ({ isMain, className, children, ...props }) => (
  <div {...props} className={css(styles.splitItem, isMain && styles.modifiers.main, className)}>
    {children}
  </div>
);

SplitItem.propTypes = propTypes;
SplitItem.defaultProps = defaultProps;

export default SplitItem;
