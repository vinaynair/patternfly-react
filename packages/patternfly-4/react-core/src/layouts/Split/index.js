export { default as Split } from './Split';
export { default as SplitItem } from './SplitItem';
