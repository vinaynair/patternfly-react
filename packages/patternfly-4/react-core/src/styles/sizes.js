export const BaseSizes = {
  xs: 'xs',
  sm: 'sm',
  md: 'md',
  lg: 'lg',
  xl: 'xl',
  '2xl': '2xl',
  '3xl': '3xl',
  '4xl': '4xl'
};

export const DeviceSizes = {
  [BaseSizes.sm]: BaseSizes.sm,
  [BaseSizes.md]: BaseSizes.md,
  [BaseSizes.lg]: BaseSizes.lg,
  [BaseSizes.xl]: BaseSizes.xl
};
