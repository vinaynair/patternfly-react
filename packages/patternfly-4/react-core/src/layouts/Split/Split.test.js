import React from 'react';
import { mount } from 'enzyme';
import Split from './Split';
import SplitItem from './SplitItem';
import { GutterSize } from '../../styles/gutters';

test('isMain', () => {
  const view = mount(
    <Split>
      <SplitItem isMain>Main content</SplitItem>
    </Split>
  );
  expect(view).toMatchSnapshot();
});

test('isMain defaults to false', () => {
  const view = mount(
    <Split>
      <SplitItem>Basic content</SplitItem>
    </Split>
  );
  expect(view).toMatchSnapshot();
});

Object.values(GutterSize).forEach(gutter => {
  test(`Gutter ${gutter}`, () => {
    const view = mount(
      <Split gutter={gutter}>
        <SplitItem>Basic Content</SplitItem>
      </Split>
    );
    expect(view).toMatchSnapshot();
  });
});
