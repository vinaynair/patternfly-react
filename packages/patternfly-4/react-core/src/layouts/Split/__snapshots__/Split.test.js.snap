// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutter lg 1`] = `
.pf-l-split__item {
  display: block;
}
.pf-l-split.pf-m-gutter {
  display: flex;
  flex-wrap: nowrap;
  padding: 0px;
  margin: 0px;
}

<Split
  className=""
  component="div"
  gutter="lg"
>
  <div
    className="pf-l-split pf-m-gutter"
  >
    <SplitItem
      className=""
      isMain={false}
    >
      <div
        className="pf-l-split__item"
      >
        Basic Content
      </div>
    </SplitItem>
  </div>
</Split>
`;

exports[`Gutter md 1`] = `
.pf-l-split__item {
  display: block;
}
.pf-l-split.pf-m-gutter {
  display: flex;
  flex-wrap: nowrap;
  padding: 0px;
  margin: 0px;
}

<Split
  className=""
  component="div"
  gutter="md"
>
  <div
    className="pf-l-split pf-m-gutter"
  >
    <SplitItem
      className=""
      isMain={false}
    >
      <div
        className="pf-l-split__item"
      >
        Basic Content
      </div>
    </SplitItem>
  </div>
</Split>
`;

exports[`Gutter sm 1`] = `
.pf-l-split__item {
  display: block;
}
.pf-l-split.pf-m-gutter {
  display: flex;
  flex-wrap: nowrap;
  padding: 0px;
  margin: 0px;
}

<Split
  className=""
  component="div"
  gutter="sm"
>
  <div
    className="pf-l-split pf-m-gutter"
  >
    <SplitItem
      className=""
      isMain={false}
    >
      <div
        className="pf-l-split__item"
      >
        Basic Content
      </div>
    </SplitItem>
  </div>
</Split>
`;

exports[`isMain 1`] = `
.pf-l-split__item.pf-m-main {
  display: block;
}
.pf-l-split {
  display: flex;
  flex-wrap: nowrap;
  padding: 0px;
  margin: 0px;
}

<Split
  className=""
  component="div"
  gutter={null}
>
  <div
    className="pf-l-split"
  >
    <SplitItem
      className=""
      isMain={true}
    >
      <div
        className="pf-l-split__item pf-m-main"
      >
        Main content
      </div>
    </SplitItem>
  </div>
</Split>
`;

exports[`isMain defaults to false 1`] = `
.pf-l-split__item {
  display: block;
}
.pf-l-split {
  display: flex;
  flex-wrap: nowrap;
  padding: 0px;
  margin: 0px;
}

<Split
  className=""
  component="div"
  gutter={null}
>
  <div
    className="pf-l-split"
  >
    <SplitItem
      className=""
      isMain={false}
    >
      <div
        className="pf-l-split__item"
      >
        Basic content
      </div>
    </SplitItem>
  </div>
</Split>
`;
